### PR TITLE
Added a feature to remove keys on reload

### PR DESCRIPTION
**Pull request by rolfyone**
Behind a feature toggle. Removes keys in the same way that the keymanager does, to ensure that the signer can't be referenced once the key has been removed.

Previously the HUP signal was purely additive, so you couldn't actually remove keys without a restart.

Made a script that allows removal of keys, so that i can add, reload, remove, reload with that interop folder.

Offers a solution to #7503

## Documentation

- [X] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [X] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

File: validator/client/src/main/java/tech/pegasys/teku/validator/client/loader/ValidatorLoader.java [hunk 1]
```diff
@@ -253,14 +264,47 @@ static ValidatorLoader create(
...
+  }
+
+  private void removeKeysNoLongerValid() {
+    final List<BLSPublicKey> currentKeySet =
+        validatorSources.stream()
+            .map(ValidatorSource::getAvailableValidators)
+            .flatMap(List::stream)
+            .map(ValidatorProvider::getPublicKey)
+            .toList();
+    final List<BLSPublicKey> removeKeys =
```
Comment (tbenr) [inline validator/client/src/main/java/tech/pegasys/teku/validator/client/loader/ValidatorLoader.java]: I'd call it `keysToBeRemoved`

File: validator/client/src/main/java/tech/pegasys/teku/validator/client/loader/ValidatorLoader.java [hunk 2]
```diff
@@ -253,14 +264,47 @@ static ValidatorLoader create(
...
         mutableLocalValidatorSource,
         mutableExternalValidatorSource,
         graffitiProvider,
         maybeDataDirLayout,
-        slashingProtectionLogger);
+        slashingProtectionLogger,
+        removeValidatorsOnReload);
+  }
+
+  private void removeKeysNoLongerValid() {
```
Comment (tbenr) [inline validator/client/src/main/java/tech/pegasys/teku/validator/client/loader/ValidatorLoader.java]: `valid` seems like too strong but I don't have a better suggestion

File: validator/client/src/main/java/tech/pegasys/teku/validator/client/loader/ValidatorLoader.java [hunk 3]
```diff
@@ -55,24 +57,32 @@ public class ValidatorLoader {
...
+
   private ValidatorLoader(
       final List<ValidatorSource> validatorSources,
       final Optional<ValidatorSource> mutableLocalValidatorSource,
       final Optional<ValidatorSource> mutableExternalValidatorSource,
       final GraffitiProvider graffitiProvider,
       final Optional<DataDirLayout> maybeDataDirLayout,
-      final SlashingProtectionLogger slashingProtectionLogger) {
+      final SlashingProtectionLogger slashingProtectionLogger,
+      final boolean remooveValidatorsOnReload) {
```
Comment (tbenr) [inline validator/client/src/main/java/tech/pegasys/teku/validator/client/loader/ValidatorLoader.java]: double `oo`

**Review by tbenr — APPROVED** [review]
LGTM. Just nits. Haven't tried the script.
Tested locally with keys on FS and works.

File: validator/client/src/main/java/tech/pegasys/teku/validator/client/loader/ValidatorLoader.java (hunk 2 above)
Comment (tbenr) [reply validator/client/src/main/java/tech/pegasys/teku/validator/client/loader/ValidatorLoader.java]: maybe `removeKeysNoLongerInConfig`, similarly to the printed log

File: validator/client/src/main/java/tech/pegasys/teku/validator/client/loader/ValidatorLoader.java (hunk 3 above)
Comment (rolfyone) [reply validator/client/src/main/java/tech/pegasys/teku/validator/client/loader/ValidatorLoader.java]: oopsie!

**Comment by rolfyone** [comment]
I broke this fairly easily, closing for now, will have to re-evaluate.

**Pull request closed without merging**